File: emergencias-backend/app/models/cotizacion.py
from sqlalchemy import Column, Integer, ForeignKey, DECIMAL, Text, String, TIMESTAMP, func
from sqlalchemy.orm import relationship
from app.database import Base

class Cotizacion(Base):
    __tablename__ = "cotizaciones"
    
    id = Column(Integer, primary_key=True, index=True)
    incidente_id = Column(Integer, ForeignKey("incidentes.id_incidente"), nullable=False)
    taller_id = Column(Integer, ForeignKey("talleres.id_taller"), nullable=False)
    precio_estimado = Column(DECIMAL(10, 2), nullable=False)
    tiempo_estimado_min = Column(Integer, nullable=False)
    descripcion = Column(Text, nullable=True)
    estado = Column(String(50), default="pendiente") # pendiente, aceptada, rechazada, expirada
    fecha_envio = Column(TIMESTAMP, server_default=func.now())
    fecha_respuesta = Column(TIMESTAMP, nullable=True)

    # Relaciones opcionales para facilitar consultas
    incidente = relationship("Incidente", backref="cotizaciones")
    taller = relationship("Taller", backref="cotizaciones")

File: movil/lib/models/cotizacion.dart
class Cotizacion {
  final int idCotizacion;
  final int incidenteId;
  final int tallerId;
  final double precioEstimado;
  final int tiempoEstimadoMin;
  final String? descripcion;
  final String estado;
  final String? tallerNombre;

  Cotizacion({
    required this.idCotizacion,
    required this.incidenteId,
    required this.tallerId,
    required this.precioEstimado,
    required this.tiempoEstimadoMin,
    this.descripcion,
    required this.estado,
    this.tallerNombre,
  });

  factory Cotizacion.fromJson(Map<String, dynamic> json) => Cotizacion(
    idCotizacion: json['id_cotizacion'],
    incidenteId: json['incidente_id'],
    tallerId: json['taller_id'],
    precioEstimado: double.tryParse(json['precio_estimado'].toString()) ?? 0,
    tiempoEstimadoMin: json['tiempo_estimado_min'] ?? 0,
    descripcion: json['descripcion'],
    estado: json['estado'] ?? 'pendiente',
    tallerNombre: json['taller_nombre'],
  );
}
